test/client/client.go

```diff
@@ -44,83 +46,124 @@ var (
 
 func init() {
 	flag.IntVar(&count, "count", 1, "Number of times to make the request")
-	flag.BoolVar(&parallel, "parallel", true, "Run requests in parallel")
-	flag.BoolVar(&insecure, "insecure", false, "Insecure requests")
-	flag.DurationVar(&timeout, "timeout", 60*time.Second, "Request timeout")
+	flag.DurationVar(&timeout, "timeout", 15*time.Second, "Request timeout")
 	flag.StringVar(&url, "url", "", "Specify URL")
 	flag.StringVar(&headerKey, "key", "", "Header key")
 	flag.StringVar(&headerVal, "val", "", "Header value")
 }
 
-func makeRequest(i int) func() error {
-	return func() error {
-		req, err := http.NewRequest("GET", url, nil)
-		if err != nil {
-			return err
-		}
+func makeHTTPRequest(client *http.Client) func(int) func() error {
+	return func(i int) func() error {
```

Comment:
I've found the nested inline functions a bit difficult to parse at times. It might be easier to read if  makeXXXRequest are defined as a normal function that takes http.Client and the iteration count and use that directly (via function pointer).

```golang
func httpRequest(client *http.Client, iter count) error {
    req, err ;= http.NewRequest("GET", url, nil)
    log.Printf("[%d] Url=%s\n", i, url)
}

var request func(client*http.Client, iter int) error
switch {
case http:
    request = httpRequest 
case grpc:
    request = grpcRequest
default: // error 
}

for i := 0; i < count; i++ {
 	g.Go(request(client, i))
}  
```